src/generic/clipboard/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,108 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { logError } from '@edx/frontend-platform/logging';
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import {
+  hideProcessingNotification,
+  showProcessingNotification,
+} from '../../processing-notification/data/slice';
+import { updateSavingStatus } from '../../data/slice';
+import { getClipboard, updateClipboard } from '../../data/api';
+import {
+  CLIPBOARD_STATUS,
+  NOTIFICATION_MESSAGES,
+  STRUCTURAL_XBLOCK_TYPES,
+  STUDIO_CLIPBOARD_CHANNEL,
+} from '../../../constants';
+import { RequestStatus } from '../../../data/constants';
+
+/**
+ * Custom React hook for managing clipboard functionality.
+ *
+ * @param canEdit - Flag indicating whether the clipboard is editable.
+ * @returns - An object containing state variables and functions related to clipboard functionality.
+ * @property showPasteUnit - Flag indicating whether the "Paste Unit" button should be visible.
+ * @property showPasteXBlock - Flag indicating whether the "Paste XBlock" button should be visible.
+ * @property sharedClipboardData - The shared clipboard data object.
+ * @property copyToClipboard - Function to copy the current selection to the clipboard.
+ */
+const useCopyToClipboard = (canEdit: boolean = true) => {
```

Comment:
Just a quick note: based on what this function does, I think `useClipboard` would be a better name. Since it reads and writes the clipboard, not just writes.